### ISSUE TITLE
Two warning tests fail

### ISSUE THREAD

**Issue by schmitts** [SALib/SALib#425]
Running `pytest` on 1.3.13 I get the following two failed tests:

```
_____________________________________________________________________________________________________________________ test_odd_num_levels_raises_warning ______________________________________________________________________________________________________________________

setup_param_file_with_groups = '/tmp/tmphtdhaumg'

    def test_odd_num_levels_raises_warning(setup_param_file_with_groups):
        parameter_file = setup_param_file_with_groups
        problem = read_param_file(parameter_file)
        with warnings.catch_warnings(record=True) as w:
            # Cause all warnings to always be triggered.
            warnings.simplefilter("always")
            # Trigger a warning.
            sample(problem, 10, num_levels=3)
            # Verify some things
>           assert len(w) == 1
E           assert 2 == 1
E             +2
E             -1

tests/sample/morris/test_morris.py:55: AssertionError
_______________________________________________________________________________________________________________________ test_even_num_levels_no_warning _______________________________________________________________________________________________________________________

setup_param_file_with_groups = '/tmp/tmpiur667kp'

    def test_even_num_levels_no_warning(setup_param_file_with_groups):
        parameter_file = setup_param_file_with_groups
        problem = read_param_file(parameter_file)
        with warnings.catch_warnings(record=True) as w:
            # Cause all warnings to always be triggered.
            warnings.simplefilter("always")
            # Trigger a warning.
            sample(problem, 10, num_levels=4)
            # Verify some things
>           assert len(w) == 0
E           assert 1 == 0
E             +1
E             -0

tests/sample/morris/test_morris.py:70: AssertionError
```

**Comment by ConnectedSystems** [comment]
Thanks for reporting this. Gotta love those comments and bare assertions...

I don't expect the failure of these tests to be high risk. It looks like these tests are just there to ensure a specific warning is thrown, but over time something else has changed causing an extra warning to be emitted as well. I will reformulate to target the specific warning only.

@willu47 I suspect this got past CI due to differences in dependency versions. Numpy especially is now throwing future/deprecation warnings for types.

**Comment by ConnectedSystems** [comment]
Confirmed this is due to recent changes to numpy:

```
{
    message : UserWarning('num_levels should be an even number, sample may be biased'), 
    category : 'UserWarning', 
    filename : 'c:\\programs\\owncloud\\projects\\salib\\src\\SALib\\sample\\morris\\morris.py', 
    lineno : 358, 
    line : None
}

{
    message : DeprecationWarning('`np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\nDeprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations'), 
    category : 'DeprecationWarning', filename : 'c:\\programs\\owncloud\\projects\\salib\\src\\SALib\\util\\__init__.py', 
    lineno : 283, 
    line : None
}

I'll fix these up now.
```

**Comment by willu47** [comment]
> Gotta love those comments and bare assertions...

Hi @ConnectedSystems  - I'd like to remind you (and all contributors) of our [Code of Conduct](https://github.com/SALib/SALib/blob/master/CODE_OF_CONDUCT.md).  Off the cuff remarks, meant in good humor may not be funny to everyone, and poorly delivered criticism can actively discourage others from contributing.  In this case, the perpetrator of these "comments and bare assertions"  was me, 7 years earlier. These are some of the first tests written for SALib, as well as some of my first tests... I would probably have the same reaction if I actually spent the time looking through the code I wrote back then, so point well received.  But please keep in mind that often an initial reaction may not be the most constructive, and particularly in an online setting, we have very little control over how our remarks and comments are interpreted.

**Comment by ConnectedSystems** [comment]
Hi @willu47 

Thanks for your reminder and for keeping me honest. Yes, it was intended in a light-hearted way and I take it you're not (too) strongly offended. I'd like to offer my apologies just the same.

**Comment by ConnectedSystems** [comment]
Issue addressed in v1.4, thanks again @schmitts for raising this.
